models/classes/task/migration/service/QueueMigrationService.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\task\migration\service;
+
+
+use common_report_Report;
+use oat\oatbox\service\ConfigurableService;
+use oat\tao\model\task\migration\StatementMigrationConfig;
+use oat\tao\model\task\migration\StatementUnit;
+use Throwable;
+
+class QueueMigrationService extends ConfigurableService
+{
+    /** @var int */
+    private $affected;
+
+    public function migrate(StatementMigrationConfig $config, ResultUnitProcessorInterface $resultUnitProcessor, common_report_Report $report): ?StatementMigrationConfig
+    {
+        $max = $this->getLastRowNumber();
+        $end = $this->calculateEndPosition(
+            $config->getStart(),
+            $config->getChunkSize(),
+            $max
+        );
+        $config->setEnd($end);
+
+        $results = $this->getResultSearcher()->search($config);
+
+        foreach ($results as $unit) {
+            try {
+                $unit = new StatementUnit($unit['subject']);
+                $resultUnitProcessor->process($unit);
```

Comment:
```suggestion
                $resultUnitProcessor->process($unit);
```
Remove statement or this is not open / closed (data source agnostic) anymore.